### PR TITLE
More GitHub action changes

### PR DESCRIPTION
**Pull request by dirkhh**
Move both the code to create atomic release numbers and the release note text into files that can be shared between multiple actions.
    
This should make the actions smaller and easier to read and since this is used in several actions it should make things much easier to maintain.

In order to test this without too much unnecessary noise, this commit only changes the android workflow - the others will be changed in a later commit once his has been tested and works (again, this can really only be tested by merging the PR into master).

### Describe the pull request:

- [x] Code cleanup
- [x] Build system change

### CODE REVIEW

**Comment by dirkhh** [comment]
lots of noise to get the syntax right... and now all the tests are pointless because only a merging the PR will run the new code. Kinda frustrating and wasteful - but I haven't figured out a better way to do this. I do think I want to see this run to completion - and I'd appreciate if you could take a look @mikeller as well.

File: gh_release_notes.in [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CICD release artifact
+
+These builds are created on every merge or push into the [Subsurface repo](http://github.com/subsurface/subsurface).
+
+This build was create by [merging PR PRNUM ("PRTITLE")](PRURL) and is based on changeset COMMITID ([link](COMMITURL))
```
Comment (mikeller) [inline gh_release_notes.in]: createD

**Review by mikeller — COMMENTED** [review]
One typo, the rest looks good.

**Comment by dirkhh** [comment]
🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️

**Comment by mikeller** [comment]
> lots of noise to get the syntax right...

You could use https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs - but this will obviously also stop the CI from running after the pull request has been merged.

**Pull request merged**